src/index.ts

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use-strict';
+
+import {
+  Contents, IContents
+} from 'jupyter-js-services';
+
+import {
+  Widget
+} from 'phosphor-widget';
+
+import './index.css';
+
+
+/**
+ * A widget which hosts a file browser.
+ */
+export
+class FileBrowser extends Widget {
+
+  /**
+   * Create a new node for the file list.
+   */
+  static createNode(): HTMLElement {
+    var node = document.createElement('div');
+    node.innerHTML = (
+      '<div class="files_inner">' +
+        '<div class="files_header">Files</div>' +
+        '<div class="list_container"></div>' +
+      '</div>'
+    );
+    return node;
+  }
+
+  /**
+   * Construct a new file browser widget.
+   */
+  constructor(baseUrl: string, currentDir: string, contents?: IContents) {
+    super();
+    this.addClass('FileBrowser');
+    this._contents = contents || new Contents(baseUrl);
+    document.addEventListener('mousedown', this, true);
+    this._currentDir = currentDir;
+  }
+
+  /**
+   * Get the onClick handler for the file browser.
+   */
+  get onClick(): (name: string, contents: any) => void {
+    return this._onClick;
+  }
+
+  /**
+   * Set the onClick handler for the file browser.
+   */
+  set onClick(cb: (name: string, contents: any) => void) {
+    this._onClick = cb;
+  }
+
+  /**
+   * Handle the DOM events for the file browser.
+   *
+   * @param event - The DOM event sent to the panel.
+   *
+   * #### Notes
+   * This method implements the DOM `EventListener` interface and is
+   * called in response to events on the panel's DOM node. It should
+   * not be called directly by user code.
+   */
+  handleEvent(event: Event): void {
```

Comment:
In other packages, `handleEvent` is usually a simple switch that deploys to private, named event handlers with names like `_evtMouseDown`. Even though you're only handling a single event type, maybe that's appropriate here?